torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -6180,3 +6180,34 @@ def test_async_execution_nested_with_cuda_future(self):
             self.assertTrue(torch.eq(fut.wait(), 3).all().item())
 
         rpc.shutdown()
+
+    @skip_if_lt_x_gpu(1)
```

Comment:
tests LGTM. 

nit: since these tests do not use RPC, does it make sense to move them to https://github.com/pytorch/pytorch/blob/master/test/test_futures.py?

Reply:
Ah! I didn't know we had such a file! But, yes, absolutely, that would be a much better home for these tests! I'll move them there in a follow-up PR, so that I can also move the previous Future tests without adding too much noise to this PR.